### PR TITLE
Use unnormalized FFT plans in Hamiltonian applies

### PR DESCRIPTION
**Pull request by antoine-levitt**
Sorry about the merge, I wasn't able to do it with a rebase

### CODE REVIEW

**Comment by antoine-levitt** [comment]
At this point, the code is perfectly aligned on the reproducer from https://github.com/JuliaMolSim/DFTK.jl/issues/349#issuecomment-721070997

File: src/PlaneWaveBasis.jl [hunk 1]
```diff
@@ -67,10 +67,15 @@ struct PlaneWaveBasis{T <: Real}
...
     # Plans for forward and backward FFT
+    # These plans follow DFTK conventions (see above)
     opFFT  # out-of-place FFT plan
     ipFFT  # in-place FFT plan
     opIFFT
     ipIFFT
+    opFFT_unnormalized   # unnormalized plans
+    ipFFT_unnormalized
+    opBFFT_unnormalized  # "backward" FFT in FFTW terminology: no normalization at all
+    ipBFFT_unnormalized
```
Comment (mfherbst) [inline src/PlaneWaveBasis.jl:78]: This starts to become a zoo. Why don't we do the normalisation in a post-step as a `./= norm` ?

Comment (mfherbst) [reply src/PlaneWaveBasis.jl:78]: Alternatively some sort of struct / named tuple(s) to hold these guys would be helpful.

Comment (antoine-levitt) [reply src/PlaneWaveBasis.jl:78]: I don't understand what you mean. This is directly used in the G_to_r! functions, so should be precomputed.

Comment (antoine-levitt) [reply src/PlaneWaveBasis.jl:78]: We could, and have a function to construct it. Can't be bothered for now though.

Comment (mfherbst) [reply src/PlaneWaveBasis.jl:78]: No I meant after the G_to_r! you use the normalisation as a post-step if normalisation is needed. So we no longer have any "scaled plans". That was my suggestion.

Comment (antoine-levitt) [reply src/PlaneWaveBasis.jl:78]: Hm, I see. I think it's nicer to have these normalized plans following DFTK conventions and the unnormalized ones as an optimization

Comment (mfherbst) [reply src/PlaneWaveBasis.jl:78]: Ok then at least split them up into two blocks of four to make it clear that the "these follow DFTK's convention" only apply to the first four blocks.

**Comment by mfherbst** [comment]
So I suppose after this it's time to rerun my scripts and see if the graphs have changed?

**Comment by antoine-levitt** [comment]
Yup! See comment in the other thread

**Comment by antoine-levitt** [comment]
Still won't get perfect scaling, but at this point this is very much diminishing returns. There are still tricks one could conceivably play but they are nasty and possibly won't improve things. I think we're at the hardware limit here.

**Comment by antoine-levitt** [comment]
Something is going very wrong with the code in fft_generic. It looks like before the line
```
    ipFFT *= sqrt(model.unit_cell_volume) / length(ipFFT)	
```
was hitting
```
*(p::DummyInplace, X) = p.fft * X
```
How can that have worked?! Any idea what we should do?

**Comment by mfherbst** [comment]
> How can that have worked?! 

I don't remember the details, but I suppose the wrapping is done by AbstractFFTs? Just provide wrappers for what is needed in `fft_generic` ?

**Comment by antoine-levitt** [comment]
No but I mean it was very buggy before, so hopefully we didn't use it...

**Comment by antoine-levitt** [comment]
So it's probably still buggy, but at least it should pass CI...

**Comment by antoine-levitt** [comment]
Hopefully somebody will pick up generic ffts again and we'll be able to drop these ugly workarounds

**Pull request merged**